libcudacxx/include/cuda/std/__cccl/rtti.h

```diff
@@ -0,0 +1,74 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef __CCCL_RTTI_H
+#define __CCCL_RTTI_H
+
+#include <cuda/std/__cccl/compiler.h>
+#include <cuda/std/__cccl/system_header.h>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+// NOTE: some compilers support the `typeid` feature but not the `dynamic_cast`
+// feature. This is why we have separate macros for each.
+
+#ifndef _CCCL_NO_RTTI
+#  if defined(CCCL_DISABLE_RTTI) // Escape hatch for users to manually disable RTTI
+#    define _CCCL_NO_RTTI
+#  elif defined(_CCCL_COMPILER_ICC)
+#    if __RTTI == 0 && __INTEL_RTTI__ == 0 && __GXX_RTTI == 0 && _CPPRTTI == 0
```

Comment:
Should that be `||` ?

Reply:
no, `&&` is correct. if any of those macros are defined, RTTI is supported.